addons/web/static/src/webclient/actions/action_service.js

```diff
@@ -694,7 +691,28 @@ export function makeActionManager(env, router = _router) {
                     if (action.target === "new") {
                         removeDialogFn?.();
                     } else {
-                        const lastCt = controllerStack[controllerStack.length - 1];
+                        let index = -1;
+                        let lastCt;
+                        index = controllerStack.findIndex((ct) => ct.jsId === controller.jsId);
+                        if (index > 0) {
+                            // The error occurred while rendering an existing controller,
+                            // so go back to the previous controller, of the current faulty one.
+                            // This occurs when clicking on the breadcrumbs.
+                            lastCt = controllerStack[index - 1];
+                            restore(lastCt.jsId);
+                            return;
+                        }
+                        index = nextStack.findIndex((ct) => ct.jsId === controller.jsId);
+                        if (index > 0) {
+                            // The error occured while rendering a new controller with a lazyController
+                            // we render this lazyController instead.
```

Comment:
not necessary, e.g. kanban > form, there, click on a many2one -> nextStack: [kanban, form, form], no lazy controller

Reply:
but you can easily detect that there's a lazy controller by comparing the nextStack and the current